Search bar release

francescanarea opened the pull request:
Fixed the implementation of search to account for partial word searches.  Simplified CSS for icons and misc search bar styling.  Tweaked search bar result styling.

francescanarea committed:
Now displays search results for all pages, not just /docs + misc design fixes

francescanarea committed:
Partial words searchable + all results displayed

francescanarea committed:
Mobile search fix

francescanarea committed:
Cleaned up icon CSS. Removed mobile search.

flowerornament commented:
@francescanarea ok, this look pretty good, but a few things to fix before release

- The magnifying glass icon isn't vertically centered in its box 
- I'd replace the "X" icon with this glyph https://rsms.me/inter/glyphs/?g=multiply.case
- I think the light gray text is too light. I see the logic, but people with poor eyesight won't be able to read it at all
- The Search box as it appears on the page before you activate it is a little too tall. Simply removing the `line-height: 1.5rem` is a good fix.

Also, not sure why titles with Urbit dates render this way ...

<img width="585" alt="Screenshot 2019-07-15 17 26 3" src="https://user-images.githubusercontent.com/31774/61257397-b2fc0480-a725-11e9-9fc1-6eed26ac821e.png">

https://urbit.org/posts/2017-4-17-update/

francescanarea committed:
remove gray on search results

francescanarea committed:
New search input reset, centered search icon, smaller search box

francescanarea committed:
Search text centered

alexmatz commented:
@msutherl we ready to go on this? How can I help get this out?

francescanarea committed:
Fix search results date formatting

Search results titles generate from document title rather than fragments of the file path.
